datalad/install.py

```diff
@@ -0,0 +1,396 @@
+        "--adjust-bashrc",
+        action="store_true",
+        help="If the scheme tweaks PATH, prepend a snippet to ~/.bashrc that exports that path.",
+    )
+    parser.add_argument(
+        "-E",
+        "--env-write-file",
+        help="Write modified environment variables to this file",
+    )
+    schemata = parser.add_subparsers(
+        title="schema",
+        dest="schema",
+        description='Type of git-annex installation (default "conda-forge")',
+    )
+    schemata.add_parser("autobuild", help="Linux, macOS only")
+    schemata.add_parser("brew", help="macOS only")
+    scm_conda_forge = schemata.add_parser("conda-forge", help="Linux only")
+    scm_conda_forge.add_argument("version", nargs="?")
+    scm_conda_forge_last = schemata.add_parser("conda-forge-last", help="Linux only")
+    scm_conda_forge_last.add_argument("version", nargs="?")
+    schemata.add_parser("datalad-extensions-build", help="Linux, macOS only")
+    scm_deb_url = schemata.add_parser("deb-url", help="Linux only")
+    scm_deb_url.add_argument("url")
+    schemata.add_parser("neurodebian", help="Linux only")
+    schemata.add_parser("neurodebian-devel", help="Linux only")
+    schemata.add_parser("snapshot", help="Linux, macOS only")
+    args = parser.parse_args()
+    if args.schema is None:
+        args.schema = "conda-forge"
+    if args.env_write_file is not None:
+        with open(args.env_write_file, "w"):
+            # Force file to exist and start out empty
+            pass
+    installer = GitAnnexInstaller(
+        adjust_bashrc=args.adjust_bashrc, env_write_file=args.env_write_file,
+    )
+    if args.schema == "autobuild":
+        installer.install_via_autobuild()
+    elif args.schema == "brew":
+        installer.install_via_brew()
+    elif args.schema == "conda-forge":
+        installer.install_via_conda_forge(args.version)
+    elif args.schema == "conda-forge-last":
+        installer.install_via_conda_forge_last(args.version)
+    elif args.schema == "datalad-extensions-build":
+        installer.install_via_datalad_extensions_build()
+    elif args.schema == "deb-url":
+        installer.install_via_deb_url(args.url)
+    elif args.schema == "neurodebian":
+        installer.install_via_neurodebian()
+    elif args.schema == "neurodebian-devel":
+        installer.install_via_neurodebian_devel()
+    elif args.schema == "snapshot":
+        installer.install_via_snapshot()
+    else:
+        raise RuntimeError(f"Invalid schema: {args.schema}")
+
+
+class GitAnnexInstaller:
+    def __init__(self, adjust_bashrc=False, env_write_file=None):
+        self._path = None
+        self.annex_bin = "/usr/bin"
+        self.adjust_bashrc = adjust_bashrc
+        self.env_write_file = Path(env_write_file)
+
+    def setpath(self, newpath):
+        self._path = newpath
+        if self.env_write_file is not None:
+            with self.env_write_file.open("a") as fp:
+                print(f"PATH={quote(newpath)}", file=fp)
```

Comment:
shouldn't we  `export PATH` ?

Reply:
also why just not to make this `addpath` and thus have here
```suggestion
                print(f"export PATH={quote(newpath)}:$PATH", file=fp)
```
so outside could `source` the file possibly from different environments (thus possibly having different PATH set).  E.g. I already used this installed with snapshots of git-annex within virtualenv and outside etc.

Reply:
Unfortunately, the conda-forge-last schema needs to add its path directory to the *end* of `PATH`.  The comment from the original source reads:

> We are interested only to get git-annex into our environment  
> So as to not interfere with "system wide" Python etc, we will add  
> miniconda at the end of the path

Reply:
that is ok, add the option `last=False` and append to `$PATH` instead of prepending `if last`

Reply:
Done.